app/views/posts/_expanded.html.erb

```diff
@@ -293,6 +293,33 @@
           </div>
         <% end %>
 
+        <% if is_question && post.children.length >= SiteSetting["TableOfContentsThreshold"] %>
+          <div class="toc" id="toc-example-toggle">
+            <button class="toc--header" data-toggle="#toc-example-toggle" data-toggle-property="class" data-toggle-value="is-active">Table of Contents</button>
+            <% post.children.each do |answer| %>
+              <a href="#" class="toc--entry">
+                <div class="toc--badge">
+                  <span class="badge is-tag is-green">+<%= answer.upvote_count %></span>
+                </div>
+                <div class="toc--badge">
+                  <span class="badge is-tag is-red">&minus;<%= answer.downvote_count %></span>
+                </div>
+                <div class="toc--full">
+                  <% sanitised = raw(sanitize(answer.body, scrubber: scrubber)) %>
```

Comment:
[**issue (blocking):**](https://conventionalcomments.org/) This isn't enough. It does prevent malicious input, but we also want to block most Markdown here. Either just send the Markdown or use the `strip_markdown` helper to strip all the markdown.

This will also make formatting easier, because the output will no longer be put into `<p>` tags, but you can fully decide that yourself.

Reply:
Seconded. Not a security issue, but you'd be better off using `strip_markdown(answer.body_markdown)[0..SiteSetting['TableOfContentsExcerptLength']]`